CONTRIBUTING.md

```diff
@@ -1,108 +1,170 @@
+5. [Adding a new supported tool](#adding-a-new-supported-tool)
 
 -----
 
-### Logging
+### Setup development environment
+From the top-level directory:
+```
+source tools/local-dev.sh
+terra
+```
+
+#### Logging
 Logging is turned off by default. Modify the root level in the `src/main/resources/logback.xml` file to turn it on (e.g. `INFO`).
 
-### Docker
-The `docker/` directory contains files required to build the Docker image.
-All files in the `scripts/` sub-directory are copied to the image, into a sub-directory that is on the `$PATH`, 
-and made executable.
+#### Troubleshooting
+- Wipe the global context directory. `rm -R $HOME/.terra`.
+- Re-run the setup script. `source tools/local-dev.sh`.
 
-The `gradle.properties` file specifies the path to the default Docker image used by the CLI.
 
-#### Build a new image
-For any change in this directory to take effect:
-1. Build a new image. This uses a short Git hash for the current commit as the tag.
+### Publish a release
+A release includes a GitHub release of the `terra-cli` repository and a corresponding Docker image pushed to GCR.
+
+To publish a release manually, from the current local code:
+1. Create a tag (e.g. `test123`) and push it to the remote repository. The tag should not include any uppercase letters.
     ```
-    > ./gradlew buildDockerImage
-    [...Docker build output...]
-    terra-cli/local:b5fdce0 successfully built
-    
-    BUILD SUCCESSFUL in 25s
-    1 actionable task: 1 executed
+    > git tag -a test123 -m "testing version 123"
+    > git push --tags
     ```
-2. Update the image id that the CLI uses. (See output of previous command for image name and tag.)
+2. Update the version in `build.gradle`.
     ```
-    > terra app set-image --image=terra-cli/local:b5fdce0
+    version = 'test123'
+    ```
+3. Login to GitHub and run the `tools/publish-release.sh` script. This will publish a pre-release, which does not
+affect the "Latest release" tag.
+    ```
+    > gh auth login
+    > ./tools/publish-release.sh test123
     ```
+    To publish a regular release, add `true` as a second argument.
+     ```
+    > gh auth login
+    > ./tools/publish-release.sh test123 true
+    ```
+
+Note that GitHub automatically attaches an archive of the source code to the release. If you have local changes that
+are not yet committed, then they may not be reflected in the source code archive, but they will be included in the
+install package. We don't use the source code archive for install.
+
+#### Install overview
+Three shell scripts are published for users.
+
+##### `download-install.sh`
+This is convenience script that:
+- Downloads the latest (or specific version) of the install package
+- Unarchives it
+- Runs the `install.sh` script included inside
+- Deletes the install package
+
+It is published as a separate file in each GitHub release.
+The intent is to have a one-line install command  `curl -L download-install.sh | bash`.
```

Comment:
Any recommendation on where to install it? I used `/usr/local/opt`. Maybe just indicate that it installs wherever you run it.

Reply:
I've been using `/usr/local/bin`. Added a note in the `README` about optionally adding it to your path post-install.